src/components/AddTodoForm.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import users from '../api/users';
+import { getUserById } from '../helpers';
+
+type Props = {
+  todo: TodoWithUser[],
+  addNewTodo: (newTodo: TodoWithUser) => void,
+};
+
+type State = {
+  newTitle: string,
+  selectedUserId: number,
+  hasTitleError: boolean,
+  hasNameError: boolean,
+};
+
+export class AddTodoForm extends React.Component<Props, State> {
+  state: State = {
+    newTitle: '',
+    selectedUserId: 0,
+    hasTitleError: false,
+    hasNameError: false,
+  };
+
+  handleNameChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      selectedUserId: +event.target.value,
+      hasNameError: false,
+    });
+  };
+
+  handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      newTitle: event.target.value,
+      hasTitleError: false,
+    });
+  };
+
+  getNewTodo = () => {
+    const { newTitle, selectedUserId } = this.state;
+    const { todo } = this.props;
+    const newTodo: TodoWithUser = {
+      id: todo.length + 1,
+      title: newTitle,
+      userId: selectedUserId,
+      user: getUserById(selectedUserId) || null,
+      completed: true,
```

Comment:
I guess newTodo should be `not completed`

```suggestion
      completed: false,
```